packages/accounts-controller/src/AccountsController.ts

```diff
@@ -759,6 +749,11 @@ export class AccountsController extends BaseController<
             },
           );
           currentState.internalAccounts.selectedAccount = accountToSelect.id;
+          currentState.internalAccounts.accounts[
+            accountToSelect.id
+          ].metadata.lastSelected = Date.now();
```

Comment:
I would encapsulate the `Date.now()` into a helper function, just to give "more context" about how we handle this "index":

```ts
/** Retrieves the index value for `metadata.lastSelected`.
  *
  * @returns The index value.
  */
function getLastSelectedIndex() {
  // NOTE: For now we use the current date, since we know this value
  // will always be higher than any already selected account index.
  return Date.now();
}

Feel free to update the doc/name for this one too! :)
```